geode-core/src/main/java/org/apache/geode/management/internal/configuration/mutators/DiskStoreManager.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.geode.management.internal.configuration.mutators;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+
+import org.apache.geode.cache.configuration.CacheConfig;
+import org.apache.geode.cache.configuration.DiskStoreType;
+import org.apache.geode.distributed.ConfigurationPersistenceService;
+import org.apache.geode.management.configuration.DiskStore;
+import org.apache.geode.management.internal.configuration.converters.DiskStoreConverter;
+
+public class DiskStoreManager extends CacheConfigurationManager<DiskStore> {
+  private final DiskStoreConverter diskStoreConverter = new DiskStoreConverter();
+
+  public DiskStoreManager(ConfigurationPersistenceService service) {
+    super(service);
+  }
+
+  @Override
+  public void add(DiskStore config, CacheConfig existing) {
+    List<DiskStoreType> diskStoreTypes = existing.getDiskStores();
+    if (diskStoreTypes.stream().noneMatch(diskStoreType -> diskStoreType.getName()
```

Comment:
I don't think you need the check here. the pipeline should already make sure that there is no existing one that has the same id exists before it's calling this `add`

Reply:
Okay, we'll check if we can drop these checks without side effects.